relentless/optimize/method.py

```diff
@@ -0,0 +1,49 @@
+import abc
+
+from .objective import ObjectiveFunction
+
+class Optimizer(abc.ABC):
+
+    @abc.abstractmethod
+    def optimize(self, objective):
+        pass
+
+class SteepestDescent(Optimizer):
+    def __init__(self, alpha, max_iter, abs_tol, mode):
+        self.alpha = alpha
+        self.max_iter = max_iter
+        self.abs_tol = abs_tol
+        self.mode = mode
+
+    def optimize(self, objective):
+        dvars = objective.design_variables()
+        iter_num = 1
+        update = {}
+
+        while True:
+            res = objective.compute()
+            for x in dvars:
+                update[x] = self.alpha*res.gradient(x)
+
+            if iter_num > self.max_iter:
```

Comment:
I don't think it should be an error to not converge... it might be better if `optimize` returned a bool saying whether the result converged or not, as the user could then decide to continue the optimization.